packages/keyring-controller/src/KeyringController.test.ts

```diff
@@ -217,20 +217,46 @@ describe('KeyringController', () => {
   });
 
   describe('createNewVaultAndKeychain', () => {
-    it('should create new vault, mnemonic and keychain', async () => {
-      const initialSeedWord = await keyringController.exportSeedPhrase(
-        password,
-      );
-      expect(initialSeedWord).toBeDefined();
-      const currentState = await keyringController.createNewVaultAndKeychain(
-        password,
-      );
-      expect(initialState).not.toBe(currentState);
-      const currentSeedWord = await keyringController.exportSeedPhrase(
-        password,
-      );
-      expect(currentSeedWord).toBeDefined();
-      expect(initialSeedWord).not.toBe(currentSeedWord);
+    describe('when there is no existing vault', () => {
+      it('should create new vault, mnemonic and keychain', async () => {
+        const cleanKeyringController = new KeyringController(
+          preferences,
+          baseConfig,
+        );
+        const initialSeedWord = await keyringController.exportSeedPhrase(
+          password,
+        );
+        const currentState =
+          await cleanKeyringController.createNewVaultAndKeychain(password);
+        const currentSeedWord = await cleanKeyringController.exportSeedPhrase(
+          password,
+        );
+        expect(initialSeedWord).toBeDefined();
+        expect(initialState).not.toBe(currentState);
+        expect(currentSeedWord).toBeDefined();
+        expect(initialSeedWord).not.toBe(currentSeedWord);
+        expect(isValidHexAddress(currentState.keyrings[0].accounts[0])).toBe(
+          true,
+        );
+      });
+    });
+
+    describe('when there is an existing vault', () => {
+      it('should return existing vault', async () => {
+        const initialSeedWord = await keyringController.exportSeedPhrase(
+          password,
+        );
+        const currentState = await keyringController.createNewVaultAndKeychain(
+          password,
+        );
+        const currentSeedWord = await keyringController.exportSeedPhrase(
```

Comment:
It looks like we are testing the results of `exportSeedPhrase` in these tests, but the `describe` block has to do with `createNewVaultAndKeychain`. I'm wondering if there is another way to test the effects of `createNewVaultAndKeychain` without depending on other methods. Or, perhaps we should test the results of `exportSeedPhrase` in the tests for that method? Out of scope for this PR, but something to think about.

Reply:
The effect of creating a new vault would be along the lines of:

* Generate a new SRP / HD keyring
* Encrypt the keyring with the given password
* Add the resulting encrypted string to the persisted store

Calling `exportSeedPhrase` does seem like a reasonably direct way to test the first two parts. It might be worth verifying that the vault was updated in the persisted store as well though.